cogs/gatekeep/gatekeep.py

```diff
@@ -0,0 +1,499 @@
+"""Birthday cog Automatically add users to a specified birthday role on their
+birthday."""
+import logging
+import os
+from random import choice
+import time  # To auto remove birthday role on the next day.
+import asyncio
+from datetime import datetime, timedelta, timezone
+import discord
+from typing import Union
+from redbot.core import Config, checks, commands, data_manager
+from redbot.core.commands.context import Context
+from redbot.core.utils import AsyncIter
+from redbot.core.utils.menus import DEFAULT_CONTROLS, menu
+from redbot.core.utils.chat_formatting import bold, pagify, spoiler, warning
+from redbot.core.bot import Red
+from .constants import *
+import string
+
+
+class Gatekeep(commands.Cog):
+    """Cog to automatically detect spam messages from accounts that meet a specific criteria.
+    An account must have joined the server less than 14 days ago. Only their first message will
+    be checked. If their first message is deemed not spam, then they will be marked safe.
+    """
+
+    def initializeConfigAndLogger(self):
+        self.config = Config.get_conf(self, identifier=5842647, force_registration=True)
+        # Register default (empty) settings.
+        self.config.register_guild(**BASE_GUILD)
+        self.config.register_member(**BASE_GUILD_MEMBER)
+
+        # Initialize logger, and save to cog folder.
+        saveFolder = data_manager.cog_data_path(cog_instance=self)
+        self.logger = logging.getLogger("red.luicogs.Gatekeep")
+        if not self.logger.handlers:
+            logPath = os.path.join(saveFolder, "info.log")
+            handler = logging.FileHandler(filename=logPath, encoding="utf-8", mode="a")
+            handler.setFormatter(
+                logging.Formatter("%(asctime)s %(message)s", datefmt="[%d/%m/%Y %H:%M:%S]")
+            )
+            self.logger.addHandler(handler)
+
+    def initializeBgTask(self):
+        # On cog load, update the watchlist daily to remove accounts older than 14 days
+        self.lastChecked = datetime.now() - timedelta(days=1)
+        self.bgTask = self.bot.loop.create_task(self.watchlistLoop())
+
+    # Class constructor
+    def __init__(self, bot: Red):
+        self.bot = bot
+        self.bgTask: asyncio.Task = None
+        self.config: Config = None
+        self.lastChecked: datetime = None
+        self.logger: logging.Logger = None
+
+        self.initializeConfigAndLogger()
+        self.initializeBgTask()
+
+    # Cancel the background task on cog unload.
+    def __unload(self):  # pylint: disable=invalid-name
+        self.bgTask.cancel()
+
+    def cog_unload(self):
+        self.__unload()
+
+    @commands.group(name="gatekeep")
+    @commands.guild_only()
+    async def _gatekeep(self, ctx: Context):
```

Comment:
Can you also decorate with `@checks.mod_or_permissions(administrator=True)`? want to avoid exposing this command tree to non-mods/admin.